addons/html_builder/static/src/builder/snippets_menu.js

```diff
@@ -124,6 +196,138 @@ export class SnippetsMenu extends Component {
+        // snippets in user databases.
+        let extraEditableZoneEls = findAll(
+            editableSavableZoneEls,
+            ".s_company_team .o_not_editable *"
+        ).filter((el) => isMediaElement(el) || el.tagName === "IMG");
+        // Same as above for social media icons.
+        const socialMediaIconEls = findAll(editableSavableZoneEls, ".s_social_media a > i");
+        extraEditableZoneEls = [...extraEditableZoneEls, ...socialMediaIconEls];
+
+        // TODO find a similar system for texts.
+        // grep: SOCIAL_MEDIA_TITLE_CONTENTEDITABLE
+        const socialMediaTitleEls = findAll(
+            editableSavableZoneEls,
+            ".s_social_media .s_social_media_title"
+        );
+        extraEditableZoneEls = [...extraEditableZoneEls, ...socialMediaTitleEls];
+
+        // To make sure the selection remains bounded to the active tab,
+        // each tab is made non editable while keeping its nested
+        // oe_structure editable. This avoids having a selection range span
+        // over all further inactive tabs when using Chrome.
+        // grep: .s_tabs
+        const tabPaneStructures = findAll(editableSavableZoneEls, ".tab-pane > .oe_structure");
+        const oeRecordCoverElements = Array.from(
+            editor.editable.querySelectorAll(
+                `${OE_RECORD_COVER_SELECTOR} [data-oe-field]:not([data-oe-field="arch"])`
+            )
+        );
+        extraEditableZoneEls = [
+            ...extraEditableZoneEls,
+            ...tabPaneStructures,
+            ...oeRecordCoverElements,
+        ];
+        return [...editableSavableZoneEls, ...extraEditableZoneEls];
+    }
+
+    /**
+     * Gets jQuery cloned element with internal text nodes escaped for XML
+     * storage.
+     *
+     * @private
+     * @param {jQuery} $el
+     * @return {jQuery}
+     */
+    getEscapedElement(el) {
+        const escapedEl = el.cloneNode(true);
+        let toEscapedEls = Array.from(escapedEl.querySelectorAll("*"));
+        toEscapedEls.push(escapedEl);
+        const filteredNotEscapedEls = toEscapedEls.filter((el) => {
+            return el.matches(
+                "object,iframe,script,style,[data-oe-model]:not([data-oe-model='ir.ui.view']"
+            );
+        });
+        const notEscapedEls = filteredNotEscapedEls.flatMap((el) =>
+            Array.from(el.querySelectorAll("*")).concat(el)
+        );
+        const toRemove = new Set(notEscapedEls);
+        toEscapedEls = toEscapedEls.filter((el) => !toRemove.has(el));
+
+        const toEscapeElChildEls = toEscapedEls.flatMap((el) => Array.from(el.childNodes));
+        toEscapeElChildEls.map((toEscapeElChild) => {
+            if (toEscapeElChild.nodeType === 3) {
+                const div = document.createElement("div");
+                div.textContent = toEscapeElChild.nodeValue;
+                toEscapeElChild.nodeValue = div.innerHTML;
+            }
+        });
+        return escapedEl;
```

Comment:
why not use import { escape } from "@web/core/utils/strings";
I may be miss something